x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/utils.ts

```diff
@@ -36,3 +39,11 @@ export class ClientError extends Error {
     this.statusCode = statusCode;
   }
 }
+
+export class DetectionRulesClientValidationError extends ZodError {
+  public readonly ruleId: string;
+  constructor(zodError: ZodError, ruleId: string) {
+    super(zodError.errors);
+    this.ruleId = ruleId;
+  }
+}
```

Comment:
Since we have two types of error classes in this utils file now, perhaps it's worth adding a short comment to both about what scenario they should be used in

Reply:
Thanks! I've added a comment for this error in [`3d37bd6`](https://github.com/elastic/kibana/pull/185748/commits/3d37bd6c45b31abe06685280546a4d93997d64de). 

As for the other error, the `ClientError`, I am still figuring out all of its use cases. I will either refactor it or add a comment to it as I convert more methods to `RuleResponse` in the upcoming PR(s).